server/website/website/tasks/async_tasks.py

```diff
@@ -403,6 +406,23 @@ def configuration_recommendation(target_data):
         except queue.Empty:
             break
 
+    # one model for each (project, session)
+    session = newest_result.session.pk
+    project = newest_result.session.project.pk
+    full_path = os.path.join(MODEL_DIR, 'p' + str(project) + '_s' + str(session) + '_nn.weights')
+
+    # neural network model
+    # FIXME: choose algorithm based on the session option
+    model_nn = NeuralNet(weights_file=full_path,
+                         n_input=X_samples.shape[1],
+                         batch_size=X_samples.shape[0],
+                         explore_iters=500,
+                         noise_scale_begin=0.1,
+                         noise_scale_end=0,
+                         debug=True)
+    model_nn.fit(X_scaled, y_scaled)
```

Comment:
Why you retrain the model with all the data every time? I have some codes in my PR which can solve this problem. I can help to rewrite this file after my PR is merged in.

Reply:
It is not retrained. It reads previous model weights every time. If it takes time, you may reduce the training iterations. But the default is 500 epochs, it only takes several seconds from my experiment.